agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -143,6 +146,19 @@ func (state *DockerTaskEngineState) ContainerMapByArn(arn string) (map[string]*a
 	return ret, ok
 }
 
+// TaskByShortID retrieves the task of a given docker short container id
+func (state *DockerTaskEngineState) TaskByShortID(cid string) (*api.Task, bool) {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	for id := range state.idToTask {
+		if strings.HasPrefix(id, cid) {
```

Comment:
I think my only (relatively minor) concern here is that it's really easy for collisions to happen (especially with super short prefixes) and it'll be non-deterministic which task gets returned (Go map traversal is randomized).  It might be worth requiring a minimum prefix length of 12 (the prefix length Docker appears to use) in order to reduce the likelihood of collisions.

@aaithal any thoughts here?